clients/client/src/main/java/org/projectnessie/client/http/impl/jdk11/JavaRequest.java

```diff
@@ -0,0 +1,204 @@
+        response = client.send(request.build(), BodyHandlers.ofInputStream());
+      } catch (HttpConnectTimeoutException e) {
+        throw new HttpClientException(
+            String.format(
+                "Timeout connecting to '%s' after %ds",
+                uri, config.getConnectionTimeoutMillis() / 1000),
+            e);
+      } catch (HttpTimeoutException e) {
+        throw new HttpClientReadTimeoutException(
+            String.format(
+                "Cannot finish %s request against '%s'. Timeout while waiting for response with a timeout of %ds",
+                method, uri, config.getReadTimeoutMillis() / 1000),
+            e);
+      } catch (MalformedURLException e) {
+        throw new HttpClientException(
+            String.format("Cannot perform %s request. Malformed Url for %s", method, uri), e);
+      } catch (IOException e) {
+        throw new HttpClientException(
+            String.format("Failed to execute %s request against '%s'.", method, uri), e);
+      } catch (InterruptedException e) {
+        throw new RuntimeException(e);
+      }
+
+      JavaResponseContext responseContext = new JavaResponseContext(response);
+
+      List<BiConsumer<ResponseContext, Exception>> callbacks = context.getResponseCallbacks();
+      if (callbacks != null) {
+        callbacks.forEach(callback -> callback.accept(responseContext, null));
+      }
+
+      config.getResponseFilters().forEach(responseFilter -> responseFilter.filter(responseContext));
+
+      if (response.statusCode() >= 400) {
+        // This mimics the (weird) behavior of java.net.HttpURLConnection.getResponseCode() that
+        // throws an IOException for these status codes.
+        throw new HttpClientException(
+            String.format(
+                "%s request to %s failed with HTTP/%d", method, uri, response.statusCode()));
+      }
+
+      response = null;
+      return new org.projectnessie.client.http.HttpResponse(responseContext, config.getMapper());
+    } finally {
+      if (response != null) {
+        try {
+          LOGGER.debug(
+              "Closing unprocessed input stream for {} request to {} delegating to {} ...",
+              method,
+              uri,
+              response.body());
+          response.body().close();
+        } catch (IOException e) {
+          // ignore
+        }
+      }
+    }
+  }
+
+  private BodyPublisher bodyPublisher(RequestContext context) {
+    ClassLoader cl = Thread.currentThread().getContextClassLoader();
+    return BodyPublishers.ofInputStream(
+        () -> {
+          try {
+            Pipe pipe = Pipe.open();
+            writerPool.execute(
+                () -> {
+                  try {
+                    // Okay - this is weird - but it is necessary when running tests with Quarkus
+                    // via `./gradlew :nessie-quarkus:test`.
+                    Thread.currentThread().setContextClassLoader(cl);
```

Comment:
Not really weird, but quite normal IMO :)

I think it'd be nice to reset the TCCL back to its original value, though.

Also, from the correctness perspective, I do think that the value of `cl` should be `body.getClass().getClassLoader()`... so it might be worth moving the TCCL logic into `writeToOutputStream`. WDYT?

Reply:
On the second though, I think it's correct to propagate current TCCL (i.e. from line 150), but I think we should still reset TCCL on the executor thread back to its original value.

Reply:
Nope. Needs to be the CL from the current thread - it's kinda weird, but makes sense (class loader loading the class !== class loader required for tests).

Reply:
On the third though :) perhaps should should use TCCL of `JavaRequest.class.getClassLoader()`.

We expect to serialize only Nessie model classes, which are already a dependency of the client code. At the same time we cannot be certain that the calling thread has the right TCCL in all cases because the thread is normally owned by engines. Also, `body.getClass().getClassLoader()` will not work it it's a `List` or model classes, for example.

Reply:
> On the third thought :) perhaps we should use TCCL of `JavaRequest.class.getClassLoader()`.

That doesn't work

Reply:
It's not a question of the "depencency" - the thread's current class loader is the one that loads needed classes.
(Tried JavaRequest + some model class - neither works)